test/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/SubsidyControllerSpec.scala

```diff
@@ -280,6 +283,303 @@ class SubsidyControllerSpec
+      }
+    }
+
+    "handling request to get returning user page" must {
+
+      def performAction = controller.getReportedPaymentReturningUserPage(
+        FakeRequest(GET, routes.SubsidyController.getReportedPaymentReturningUserPage.url)
+      )
+
+      "call to get session fails" in {
+        inSequence {
+          mockAuthWithEnrolmentAndValidEmail()
+          mockRetrieveUndertaking(eori1)(undertaking.some.toFuture)
+          mockGetOrCreate[SubsidyJourney](eori1)(Left(ConnectorError(exception)))
+        }
+        assertThrows[Exception](await(performAction))
+      }
+
+      "display the page" when {
+        "a request is made" in {
+          inSequence {
+            mockAuthWithEnrolmentAndValidEmail()
+            mockRetrieveUndertaking(eori1)(undertaking.some.toFuture)
+            mockGetOrCreate[SubsidyJourney](eori1)(Right(subsidyJourney))
+          }
+          checkPageIsDisplayed(
+            performAction,
+            messageFromMessageKey("reportPaymentReturnJourney.title"),
+            { doc =>
+              doc.select(".govuk-back-link").attr("href") shouldBe routes.AccountController.getAccountPage.url
+              doc
+                .select("form")
+                .attr("action") shouldBe routes.SubsidyController.postReportedPaymentReturningUserPage.url
+            }
+          )
+        }
+      }
+    }
+
+    "handling get to reported no custom subsidy page" must {
+      def performAction(): Future[Result] = controller.getReportedNoCustomSubsidyPage(FakeRequest())
+
+      "render the page with updated form and correct URLs" in {
+        val journeyWithReportedNonCustomSubsidyReturningUser = subsidyJourney
+          .setReportedPaymentReturningUser(true)
+
+        inSequence {
+          mockAuthWithEnrolmentAndValidEmail(eori1)
+          mockRetrieveUndertaking(eori1)(undertaking.some.toFuture)
+          mockGetOrCreate[SubsidyJourney](eori1)(Right(journeyWithReportedNonCustomSubsidyReturningUser))
+        }
+
+        val result = performAction()
+        val document: Document = Jsoup.parse(contentAsString(result))
+
+        val expectedTaxYears = List(
+          (LocalDate.of(2020, 4, 6), LocalDate.of(2021, 4, 5)),
+          (LocalDate.of(2021, 4, 6), LocalDate.of(2022, 4, 5)),
+          (LocalDate.of(2022, 4, 6), LocalDate.of(2023, 4, 5))
+        )
+
+        val formatter = DateTimeFormatter.ofPattern("d MMMM yyyy")
+
+        val listItems = document.select("ul.govuk-list li")
+
+        status(result) shouldBe OK
+
+        document
+          .select(".govuk-back-link")
+          .attr("href") shouldBe routes.SubsidyController.getReportedPaymentReturningUserPage.url
+
+        listItems.size shouldBe expectedTaxYears.size
+
+        expectedTaxYears.zipWithIndex.foreach { case ((startYear, endYear), index) =>
+          val expectedRange = s"${startYear.format(formatter)} to ${endYear.format(formatter)}"
+          listItems.get(index).text() should include(expectedRange)
```

Comment:
would be better to do "shouldBe" rather than "should include" (just in case) there is extra text in the element.

Reply:
also, it might be a lot easier and clearer to just add ids to the "li" tags and do `document.getElementById` to select the element and then check the content